src/main/java/org/gooru/insights/services/MailServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package org.gooru.insights.services;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.MailParseException;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MailServiceImpl implements MailService {
+
+	@Autowired
+	private JavaMailSender mailSender;
+
+	@Autowired
+	private SimpleMailMessage preConfiguredMessage;
+	
+	@Autowired
+	private JavaMailSender defaultMailSender;
+
+	/**
+	 * This method will send compose and send the message
+	 * */
+	public void sendMail(String to, String subject, String body, String file) {
```

Comment:
sendDefaultMail, sendmail functions  are doing  the same logic expect the recipient   address change.